pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go

```diff
@@ -135,6 +132,31 @@ func (nc *NamespaceController) Run(stopCh <-chan struct{}) {
 	}
 	log.Infof("Namespace controller started")
 	go nc.queue.Run(stopCh)
+	go nc.startCaBundleWatcher(stopCh)
+}
+
+// startCaBundleWatcher listens for updates to the CA bundle and update cm in each namespace
+func (nc *NamespaceController) startCaBundleWatcher(stop <-chan struct{}) {
+	id, watchCh := nc.caBundleWatcher.AddWatcher()
```

Comment:
nit:
```suggestion
	watchCh, cancel := nc.caBundleWatcher.AddWatcher()
	defer cancel()
```

May be more idiomatic interface